shopinvader_partner_vat/services/vat.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Akretion (http://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.addons.component.core import Component
+
+import logging
+_logger = logging.getLogger(__name__)
+
+try:
+    import stdnum.eu.vat
+except (ImportError, IOError) as err:
+        _logger.debug(err)
+
+
+class CustomerService(Component):
+    _inherit = 'shopinvader.customer.service'
+
+    def check_vat(self, **params):
+        partner_obj = self.env['res.partner']
+        country_code, vat_number = partner_obj._split_vat(params['vat_number'])
+        vat_number = country_code.upper() + vat_number
+        res = {
+            'valid': partner_obj.simple_vat_check(country_code, vat_number),
+            'vat_number': vat_number,
+            }
+        if self.shopinvader_backend.company_id.vat_check_vies and\
+                self.env['res.country'].search([
+                    ('code', '=ilike', country_code),
+                    ('country_group_ids', '=',
+                        self.env.ref('base.europe').id)]):
+            try:
+                response = stdnum.eu.vat.check_vies(vat_number)
+                if response['valid']:
+                    res.update({
+                        'with_details': True,
+                        'name': response['name'],
+                        'address': response['address'],
+                        'valid': True,
+                        })
+                else:
+                    res['valid'] = False
+            except Exception:
+                # TODO improve exception management
+                _logger.debug('Invalid number for vies')
+        return {'data': res}
```

Comment:
@sebastienbeau Why do you wrap the response into a 'data' value? it should be no more required by the locomotive proxy.  See proposed changes https://github.com/akretion/locomotive_shopinvader/pull/14/commits/eb16dc300a30a8d8d3786d0069f2fd4ca7e2c41d#diff-784052589f54df8a8c51c7af2a958664L92